services/src/main/java/org/apache/druid/cli/CliPeon.java

```diff
@@ -303,19 +304,7 @@ public void configure(Binder binder)
           @Named(ServiceStatusMonitor.HEARTBEAT_TAGS_BINDING)
           public Supplier<Map<String, Object>> heartbeatDimensions(Task task)
           {
-            ImmutableMap.Builder<String, Object> builder = ImmutableMap.builder();
-            builder.put(DruidMetrics.TASK_ID, task.getId());
-            builder.put(DruidMetrics.DATASOURCE, task.getDataSource());
-            builder.put(DruidMetrics.TASK_TYPE, task.getType());
-            builder.put(DruidMetrics.GROUP_ID, task.getGroupId());
-            Map<String, Object> tags = task.getContextValue(DruidMetrics.TAGS);
-            if (tags != null && !tags.isEmpty()) {
-              builder.put(DruidMetrics.TAGS, tags);
-            }
-
-            return Suppliers.ofInstance(
-                builder.build()
-            );
+            return Suppliers.ofInstance(CliPeon.heartbeatDimensions(task));
```

Comment:
Does this mean that the method `heartbeatDimensions` method would be called just once when initializing?
Don't we want the `status` and the `heartbeatDimensions` to be freshly computed each time the metric is emitted?

Reply:
Good catch!! I think this should be fixed now.